OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -67,6 +69,15 @@ public DecisionService(Bucketer bucketer, IErrorHandler errorHandler, ProjectCon
         /// <returns>The Variation the user is allocated into.</returns>
         public Variation GetVariation(Experiment experiment, string userId, UserAttributes filteredAttributes)
         {
+            string bucketingId = userId;
+
+            // If the bucketing ID key is defined in attributes, then use that in place of the userID for the murmur hash key
+            if (!string.IsNullOrEmpty(filteredAttributes[RESERVED_ATTRIBUTE_KEY_BUCKETING_ID]))
```

Comment:
it may cause exception if the key is not present, plz. use contains method

Reply:
Done